code/datums/dna.dm

```diff
@@ -1,3 +1,41 @@
+///Setter macro used to modify dna blocks. For convenience, use SET_UI_BLOCK for identity and SET_UF_BLOCK for features.
+#define SET_DNA_BLOCK(string, blocknum, input, len_by_block) string = copytext(string, 1, len_by_block[blocknum]) + input + (blocknum < len_by_block.len ? copytext(string, len_by_block[blocknum+1]) : 0)
+
+#define SET_UI_BLOCK(string, blocknum, input) SET_DNA_BLOCK(string, blocknum, input, GLOB.total_ui_len_by_block)
+#define SET_UF_BLOCK(string, blocknum, input) SET_DNA_BLOCK(string, blocknum, input, GLOB.total_uf_len_by_block)
```

Comment:
Why are these macros? it's not a horribly hot run of code right? so it's just kinda spooking off newcoders who don't know how to ctrl click something and making things harder to read (I've had to explain QDEL_NULL more times then I'd like to admit)

Reply:
You didn't respond to this

Reply:
> Why are these macros? it's not a horribly hot run of code right?

It's not a horribly hot run piece of code, just a fairly bad micro op. If I didn't repond before, that's because I aknowledged your point and saw no reason to respond, thinking the questions were rhetorical.